### PR TITLE
13.0 mig account cutoff base

### PR DESCRIPTION
**Pull request by jimhoefnagels**

### CODE REVIEW

**Comment by TDu** [comment]
LG :+1:

File: account_cutoff_base/models/account_cutoff.py [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+        label = self.cutoff_type_label_map.get(cutoff_type, "")
+        return label
+
+    def _selection_cutoff_type(self):
+        # generate cutoff types from mapping
+        return list(self.cutoff_type_label_map.items())
+
+    @api.model
+    def _default_cutoff_account_id(self):
+        """Default account muast always be None"""
```
Comment (i-vyshnevska) [inline account_cutoff_base/models/account_cutoff.py]:
```suggestion
        """Default account must always be None"""
```

File: account_cutoff_base/models/account_cutoff.py [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+    @api.model
+    def _default_cutoff_journal_id(self):
+        return self.env.user.company_id.default_cutoff_journal_id
+
+    cutoff_date = fields.Date(
+        string="Cut-off Date",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        copy=False,
+        track_visibility="onchange",
```
Comment (i-vyshnevska) [inline account_cutoff_base/models/account_cutoff.py]:
```suggestion
        tracking=True,
```
api of mail tred changed

File: account_cutoff_base/models/account_cutoff.py [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+        default=lambda self: self._default_cutoff_journal_id(),
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    total_cutoff_amount = fields.Monetary(
+        compute="_compute_total_cutoff",
+        string="Total Cut-off Amount",
+        currency_field="company_currency_id",
+        readonly=True,
+        track_visibility="onchange",
```
Comment (i-vyshnevska) [inline account_cutoff_base/models/account_cutoff.py]:
```suggestion
        tracking=True,,
```

File: account_cutoff_base/models/account_cutoff.py [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+        string="Company",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default=lambda self: self.env["res.company"]._company_default_get(
+            "account.cutoff"
+        ),
+    )
+    company_currency_id = fields.Many2one(
+        related="company_id.currency_id", readonly=True, string="Company Currency"
```
Comment (i-vyshnevska) [inline account_cutoff_base/models/account_cutoff.py]:
```suggestion
        related="company_id.currency_id", string="Company Currency"
```
already readonly

File: account_cutoff_base/models/account_cutoff.py [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+        inverse_name="parent_id",
+        string="Cut-off Lines",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    state = fields.Selection(
+        selection=[("draft", "Draft"), ("done", "Done")],
+        index=True,
+        readonly=True,
+        track_visibility="onchange",
```
Comment (i-vyshnevska) [inline account_cutoff_base/models/account_cutoff.py]:
```suggestion
        tracking=True,
```

**Review by TDu — APPROVED** [review]

**Pull request merged**